app/src/validators/common.js

```diff
@@ -45,11 +45,12 @@ const type = {
     version: 'uuidv4'
   }),
 
-  metadata: (min) => Joi.object()
-    .pattern(/^x-amz-meta-.{1,255}$/i, Joi.string().min(min).max(255))
+  metadata: (min = 0) => Joi.object()
+    .pattern(/^x-amz-meta-.{1,255}$/i, Joi.string().min(1).max(255), { matches: Joi.array().min(min) })
     .unknown(),
 
-  tagset: (min) => Joi.object().pattern(/^.{1,128}$/, Joi.string().min(min).max(255))
+  tagset: (min = 0) => Joi.object()
+    .pattern(/^.{1,128}$/, Joi.string().min(1).max(255), { matches: Joi.array().min(min) })
```

Comment:
Tag values can be up to [256 characters](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html)

Reply:
Hey Phil. Glad to see you're still following the repo.
Good to know about the character limit. I think we limited this to 255 because the field in the db where we store this value is limited to 255. Next time we do a db migration we'll consider increasing. Agree, it does seem a little odd in comparison to the S3 docs.

Reply:
It is also documented as 256 in your [Metadata-Tag](https://github.com/bcgov/common-object-management-service/wiki/Metadata-Tag#tag) wiki page.

Reply:
We've added this discrepancy to our internal backlog and will address it before the v0.4.0 release.

Reply:
I would also comment, that metadata keys must conform to the [valid http header spec](https://www.rfc-editor.org/rfc/rfc2616#section-4.2). Cant contain control characters, or any of: ()<>@,;:\"/[]?={} _space_ and _tab_